src/main/java/com/nike/cerberus/cli/EnvironmentConfigToArgsMapper.java

```diff
@@ -86,13 +80,36 @@ private EnvironmentConfigToArgsMapper() {
         }
 
         // convert to string array and return
-        return args.toArray(new String[args.size()]);
+        return args.toArray(new String[0]);
     }
 
     public static List<String> getArgsForCommand(EnvironmentConfig environmentConfig, String commandName, String[] passedArgs) {
         if (environmentConfig == null) {
             return Lists.newArrayList(passedArgs);
         }
+        return getArgsForCommand(environmentConfig, commandName, passedArgs, environmentConfig.getPrimaryRegion());
+    }
+
+    public static List<String> getArgsForCommand(EnvironmentConfig environmentConfig, String commandName, String[] passedArgs, Optional<String> region) {
```

Comment:
This one optionally takes the region and defaults to the primary region where as the other explicitly expects the region to be passed.

Reply:
But if both of those fourth arguments are Strings, which function will be called in this example?

.. = getArgsForCommand(environmentConfig, commandName, passedArgs, "us-west-2");

Or does it not matter, because the outcome will be the same either way?

Reply:
One takes a String the other takes a String Optional, those are different method signatures.

Reply:
One takes a String the other takes a String Optional, those are different method signatures.